kv/kvnemesis: TestKVNemesisMultiNode failed

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13801738?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13801738?buildTab=artifacts#/) on release-22.2 @ [6d16c97c1b9f553b81fb36200106bc6c5665ab3a](https://github.com/cockroachdb/cockroach/commits/6d16c97c1b9f553b81fb36200106bc6c5665ab3a):


```
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1594
        Wraps: (2) error applying x.BarrierWithLAI(ctx, /Table/100, /Table/100/"e0ffec28") // key range /Table/100-/Table/100/"e0ffec28" outside of bounds of range /Table/52-/Table/100/"3fd38309"; suggested ranges: [desc: r52:/Table/{52-100/"3fd38309"} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n3,s3):4, next=5, gen=14], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Wraps: (3) key range /Table/100-/Table/100/"e0ffec28" outside of bounds of range /Table/52-/Table/100/"3fd38309"; suggested ranges: [desc: r52:/Table/{52-100/"3fd38309"} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n3,s3):4, next=5, gen=14], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.RangeKeyMismatchError
    kvnemesis_test.go:111: failure:
        error applying x.BarrierWithLAI(ctx, /Table/100/"c49da1bf", /Table/101) // key range /Table/100/"c49da1bf"-/Table/101 outside of bounds of range /Table/100/"c49da1bf"-/Table/100/"e0ffec28"; suggested ranges: [desc: r103:/Table/100/"{c49da1bf"-e0ffec28"} [(n2,s2):1, (n3,s3):2, (n4,s4):3, next=4, gen=38, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]: key range /Table/100/"c49da1bf"-/Table/101 outside of bounds of range /Table/100/"c49da1bf"-/Table/100/"e0ffec28"; suggested ranges: [desc: r103:/Table/100/"{c49da1bf"-e0ffec28"} [(n2,s2):1, (n3,s3):2, (n4,s4):3, next=4, gen=38, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:894
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:444
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:69
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis.go:97
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:107
          | testing.tRunner
          | 	GOROOT/src/testing/testing.go:1446
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1594
        Wraps: (2) error applying x.BarrierWithLAI(ctx, /Table/100/"c49da1bf", /Table/101) // key range /Table/100/"c49da1bf"-/Table/101 outside of bounds of range /Table/100/"c49da1bf"-/Table/100/"e0ffec28"; suggested ranges: [desc: r103:/Table/100/"{c49da1bf"-e0ffec28"} [(n2,s2):1, (n3,s3):2, (n4,s4):3, next=4, gen=38, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Wraps: (3) key range /Table/100/"c49da1bf"-/Table/101 outside of bounds of range /Table/100/"c49da1bf"-/Table/100/"e0ffec28"; suggested ranges: [desc: r103:/Table/100/"{c49da1bf"-e0ffec28"} [(n2,s2):1, (n3,s3):2, (n4,s4):3, next=4, gen=38, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.RangeKeyMismatchError
    kvnemesis_test.go:111: failure:
        error applying x.BarrierWithLAI(ctx, /Table/100/"3dc891b8", /Table/100/"5f50e36f") // key range /Table/100/"3dc891b8"-/Table/100/"5f50e36f" outside of bounds of range /Table/100/"3dc891b8"-/Table/100/"3fd38309"; suggested ranges: [desc: r94:/Table/100/"3{dc891b8"-fd38309"} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n3,s3):4, next=5, gen=23, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]: key range /Table/100/"3dc891b8"-/Table/100/"5f50e36f" outside of bounds of range /Table/100/"3dc891b8"-/Table/100/"3fd38309"; suggested ranges: [desc: r94:/Table/100/"3{dc891b8"-fd38309"} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n3,s3):4, next=5, gen=23, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:894
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:444
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:69
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis.go:97
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:107
          | testing.tRunner
          | 	GOROOT/src/testing/testing.go:1446
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1594
        Wraps: (2) error applying x.BarrierWithLAI(ctx, /Table/100/"3dc891b8", /Table/100/"5f50e36f") // key range /Table/100/"3dc891b8"-/Table/100/"5f50e36f" outside of bounds of range /Table/100/"3dc891b8"-/Table/100/"3fd38309"; suggested ranges: [desc: r94:/Table/100/"3{dc891b8"-fd38309"} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n3,s3):4, next=5, gen=23, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Wraps: (3) key range /Table/100/"3dc891b8"-/Table/100/"5f50e36f" outside of bounds of range /Table/100/"3dc891b8"-/Table/100/"3fd38309"; suggested ranges: [desc: r94:/Table/100/"3{dc891b8"-fd38309"} [(n1,s1):1, (n2,s2):2, (n4,s4):3, (n3,s3):4, next=5, gen=23, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.RangeKeyMismatchError
    kvnemesis_test.go:113: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1f42cf5be2fc021646bf9b2daf5eaef3/logTestKVNemesisMultiNode2400693334
--- FAIL: TestKVNemesisMultiNode (13.39s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118471 kv/kvnemesis: TestKVNemesisMultiNode failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-23.2]
- #118017 kv/kvnemesis: TestKVNemesisMultiNode failed [A-testing C-bug C-test-failure O-robot P-2 T-kv branch-master release-blocker]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35818

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13801739?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13801739?buildTab=artifacts#/) on release-22.2 @ [6d16c97c1b9f553b81fb36200106bc6c5665ab3a](https://github.com/cockroachdb/cockroach/commits/6d16c97c1b9f553b81fb36200106bc6c5665ab3a):


```
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1594
        Wraps: (2) error applying x.BarrierWithLAI(ctx, /Table/100/"3bc99311", /Table/100/"5aedde52") // key range /Table/100/"3bc99311"-/Table/100/"5aedde52" outside of bounds of range /Table/100/"3bc99311"-/Table/100/"5007e82a"; suggested ranges: [desc: r57:/Table/100/"{3bc99311"-5007e82a"} [(n1,s1):1, (n3,s3):2, next=3, gen=9, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Wraps: (3) key range /Table/100/"3bc99311"-/Table/100/"5aedde52" outside of bounds of range /Table/100/"3bc99311"-/Table/100/"5007e82a"; suggested ranges: [desc: r57:/Table/100/"{3bc99311"-5007e82a"} [(n1,s1):1, (n3,s3):2, next=3, gen=9, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.RangeKeyMismatchError
    kvnemesis_test.go:111: failure:
        error applying x.BarrierWithLAI(ctx, /Table/100/"7fb95cf2", /Table/100/"9a4e1102") // key range /Table/100/"7fb95cf2"-/Table/100/"9a4e1102" outside of bounds of range /Table/100/"7bd1e4cf"-/Table/100/"8a3f7d84"; suggested ranges: [desc: r55:/Table/100/"{7bd1e4cf"-8a3f7d84"} [(n1,s1):1, (n2,s2):2, next=3, gen=11, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]: key range /Table/100/"7fb95cf2"-/Table/100/"9a4e1102" outside of bounds of range /Table/100/"7bd1e4cf"-/Table/100/"8a3f7d84"; suggested ranges: [desc: r55:/Table/100/"{7bd1e4cf"-8a3f7d84"} [(n1,s1):1, (n2,s2):2, next=3, gen=11, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:894
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:444
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:69
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis.go:97
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:107
          | testing.tRunner
          | 	GOROOT/src/testing/testing.go:1446
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1594
        Wraps: (2) error applying x.BarrierWithLAI(ctx, /Table/100/"7fb95cf2", /Table/100/"9a4e1102") // key range /Table/100/"7fb95cf2"-/Table/100/"9a4e1102" outside of bounds of range /Table/100/"7bd1e4cf"-/Table/100/"8a3f7d84"; suggested ranges: [desc: r55:/Table/100/"{7bd1e4cf"-8a3f7d84"} [(n1,s1):1, (n2,s2):2, next=3, gen=11, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Wraps: (3) key range /Table/100/"7fb95cf2"-/Table/100/"9a4e1102" outside of bounds of range /Table/100/"7bd1e4cf"-/Table/100/"8a3f7d84"; suggested ranges: [desc: r55:/Table/100/"{7bd1e4cf"-8a3f7d84"} [(n1,s1):1, (n2,s2):2, next=3, gen=11, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.RangeKeyMismatchError
    kvnemesis_test.go:111: failure:
        error applying x.BarrierWithLAI(ctx, /Table/100/"7fb95cf2", /Table/100/"9a4e1102") // key range /Table/100/"7fb95cf2"-/Table/100/"9a4e1102" outside of bounds of range /Table/100/"7fb95cf2"-/Table/100/"8a3f7d84"; suggested ranges: [desc: r79:/Table/100/"{7fb95cf2"-8a3f7d84"} [(n1,s1):1, (n2,s2):2, next=3, gen=12, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]: key range /Table/100/"7fb95cf2"-/Table/100/"9a4e1102" outside of bounds of range /Table/100/"7fb95cf2"-/Table/100/"8a3f7d84"; suggested ranges: [desc: r79:/Table/100/"{7fb95cf2"-8a3f7d84"} [(n1,s1):1, (n2,s2):2, next=3, gen=12, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).failIfError
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:894
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:444
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:69
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis.go:97
          | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestKVNemesisMultiNode
          | 	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/kvnemesis_test.go:107
          | testing.tRunner
          | 	GOROOT/src/testing/testing.go:1446
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_amd64.s:1594
        Wraps: (2) error applying x.BarrierWithLAI(ctx, /Table/100/"7fb95cf2", /Table/100/"9a4e1102") // key range /Table/100/"7fb95cf2"-/Table/100/"9a4e1102" outside of bounds of range /Table/100/"7fb95cf2"-/Table/100/"8a3f7d84"; suggested ranges: [desc: r79:/Table/100/"{7fb95cf2"-8a3f7d84"} [(n1,s1):1, (n2,s2):2, next=3, gen=12, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Wraps: (3) key range /Table/100/"7fb95cf2"-/Table/100/"9a4e1102" outside of bounds of range /Table/100/"7fb95cf2"-/Table/100/"8a3f7d84"; suggested ranges: [desc: r79:/Table/100/"{7fb95cf2"-8a3f7d84"} [(n1,s1):1, (n2,s2):2, next=3, gen=12, sticky=9223372036.854775807,2147483647], lease: <empty>, closed_timestamp_policy: LAG_BY_CLUSTER_SETTING]
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *roachpb.RangeKeyMismatchError
    kvnemesis_test.go:113: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1f42cf5be2fc021646bf9b2daf5eaef3/logTestKVNemesisMultiNode303392942
--- FAIL: TestKVNemesisMultiNode (32.83s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118471 kv/kvnemesis: TestKVNemesisMultiNode failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-23.2]
- #118017 kv/kvnemesis: TestKVNemesisMultiNode failed [A-testing C-bug C-test-failure O-robot P-2 T-kv branch-master release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvnemesis/validator.go
+++ b/pkg/kv/kvnemesis/validator.go
@@ -434,7 +434,7 @@ func (v *validator) processOp(txnID *string, op Operation) {
 		}
 	case *BarrierOperation:
 		if op.Barrier.WithLeaseAppliedIndex &&
-			resultIsError(t.Result, &roachpb.RangeKeyMismatchError{}) {
+			errors.HasType(errorFromResult(t.Result), (*roachpb.RangeKeyMismatchError)(nil)) {
 			// Barriers requesting LAIs can't span ranges. The generator will
 			// optimistically try to fit the barrier inside one of the current ranges,
 			// but this may race with a split, so we ignore the error in this case and

